forge/test/mlir/test_ops.py

```diff
@@ -1032,3 +1032,65 @@ def forward(self, x):
     co_out = [co.to("cpu") for co in co_out]
     fw_out = [fw_out] if isinstance(fw_out, torch.Tensor) else fw_out
     assert all([compare_with_golden(golden=fo, calculated=co, pcc=0.99) for fo, co in zip(fw_out, co_out)])
+
+
+@pytest.mark.parametrize("shape", [(1, 3, 32, 32)])
+@pytest.mark.parametrize(
+    "padding",
+    [
+        (1, 1, 1, 1),
+        (1, 1, 2, 2),
+    ],
+)
+@pytest.mark.xfail(reason="'ttnn.conv2d' op Bias must only have data on the final dimenstion")
+def test_padding_with_conv2d(shape, padding):
+    class PaddingAndConv2d(nn.Module):
+        def __init__(self, padding):
+            super().__init__()
+            self.padding = padding
+            self.conv = nn.Conv2d(3, 3, kernel_size=3, stride=1, padding=0)
+
+        def forward(self, x):
+            x = nn.functional.pad(x, self.padding, mode="constant", value=0)
+            return self.conv(x)
+
+    framework_model = PaddingAndConv2d(padding=padding)
+
+    inputs = [torch.rand(shape)]
+    compiled_model = forge.compile(framework_model, sample_inputs=inputs)
+    co_out = compiled_model(*inputs)
+
+    co_out = [co.to("cpu") for co in co_out]
+    fw_out = [fw_out] if isinstance(fw_out, torch.Tensor) else fw_out
+    assert all([compare_with_golden(golden=fo, calculated=co, pcc=0.99) for fo, co in zip(fw_out, co_out)])
+
+
+@pytest.mark.parametrize("shape", [(1, 3, 32, 32)])
+@pytest.mark.parametrize(
+    "padding",
+    [
+        (1, 1, 1, 1),
+        (1, 1, 2, 2),
+    ],
+)
+@pytest.mark.xfail(reason="Only BFLOAT16 supported for now")
```

Comment:
Can we use bfloat16 in this test instead of marking it as xfailed?

Reply:
On converting both framework_model and inputs to torch.bfloat16,

- Testcase 1 hit with `For valid non-interleaved buffers page size 22 must equal buffer size 724. For interleaved-buffers page size should be divisible by buffer size`
- Testcase 2 hit with `Page size must be divisible by sizeof(uint32_t) because buffers hold uint32_t values`

Shall I create issues respectively for both issues @nvukobratTT ?

### Logs:
[test_padding_with_maxpool2d.log](https://github.com/user-attachments/files/17630509/test_padding_with_maxpool2d.log)